spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -80,10 +93,15 @@ protected void onInit() {
 
 	/**
 	 * Takes no action by default. Subclasses may override this if they
-	 * need lifecycle-managed behavior.
+	 * need lifecycle-managed behavior. Protected by 'lifecycleLock'.
 	 */
 	@Override
 	protected void doStart() {
+		Assert.state(this.outputChannel != null || StringUtils.hasText(this.outputChannelName),
+				"'outputChannel' or 'outputChannelName' is required");
+		if (this.outputChannelName != null) {
+			this.outputChannel = getChannelResolver().resolveDestination(this.outputChannelName);
+		}
```

Comment:
No; I decided to leave it off - we allow changing the `outputChannel` between stop/start so we should do the same for `outputChannelName`.